sourced/ml/transformers/basic.py

```diff
@@ -134,17 +136,28 @@ def __init__(self, save_loc, **kwargs):
 
     def __call__(self, df: DataFrame):
         if self.explained:
+            self._log.info("toDebugString():\n%s", df.explain(True).decode())
             self._log.info("toDebugString():\n%s", df.rdd.toDebugString().decode())
         df.write.parquet(self.save_loc)
 
 
 class ParquetLoader(Transformer):
-    def __init__(self, session, **kwargs):
+    def __init__(self, session, paths, **kwargs):
         super().__init__(**kwargs)
         self.session = session
+        self.paths = paths
 
-    def __call__(self, df):
-        return self.session.read.parquet(self.save_loc)
+    def __getstate__(self):
+        state = super().__getstate__()
+        del state["session"]
+        return state
+
+    def __call__(self, _):
+        if issubclass(type(self.paths), list):
+            return self.session.read.parquet(*self.paths)
+        elif issubclass(type(self.paths), str):
```

Comment:
You already returned, no `elif` is needed

Reply:
according to http://spark.apache.org/docs/2.2.0/api/python/pyspark.sql.html#pyspark.sql.DataFrameReader.parquet it can be either list of str (it's a bit confusing in their documentation)